webapp/channels/src/components/access_history_modal/access_history_modal.test.tsx

```diff
@@ -38,26 +40,23 @@ describe('components/AccessHistoryModal', () => {
         expect(wrapper.find(AuditTable).exists()).toBe(true);
     });
 
-    test('should have called actions.getUserAudits when onShow is called', () => {
+    test('should have called actions.getUserAudits only when first rendered', () => {
         const actions = {
             getUserAudits: jest.fn(),
         };
         const props = {...baseProps, actions};
-        const wrapper = shallow<AccessHistoryModal>(
-            <AccessHistoryModal {...props}/>,
-        );
+        const view = render(withIntl(<AccessHistoryModal {...props}/>));
 
-        wrapper.instance().onShow();
-        expect(actions.getUserAudits).toHaveBeenCalledTimes(2);
+        expect(actions.getUserAudits).toHaveBeenCalledTimes(1);
+        const newProps = {...props, currentUserId: 'foo'};
+        view.rerender(withIntl(<AccessHistoryModal {...newProps}/>));
+        expect(actions.getUserAudits).toHaveBeenCalledTimes(1);
     });
 
-    test('should match state when onHide is called', () => {
-        const wrapper = shallow<AccessHistoryModal>(
-            <AccessHistoryModal {...baseProps}/>,
-        );
-
-        wrapper.setState({show: true});
-        wrapper.instance().onHide();
-        expect(wrapper.state('show')).toEqual(false);
+    test('should hide', async () => {
+        render(withIntl(<AccessHistoryModal {...baseProps}/>));
+        await waitFor(() => screen.getByText('Access History'));
+        fireEvent.click(screen.getByLabelText('Close'));
+        await waitForElementToBeRemoved(() => screen.getByText('Access History'));
```

Comment:
The previous test was testing internal state, which is not the best practice. Here we are testing with internals of the children component, which is also not the best practice. Either way, both test the same behavior.